multiqc/modules/dragen/coverage_metrics.py

```diff
@@ -355,83 +390,780 @@ def _create_table(self, data_by_sample, own_tabl_headers, table_name: str, table
             5. Reads with `MAPQ` = `0` (multimappers)
             6. Overlapping mates are double-counted
             """,
-            plot=table.plot(data_by_sample, own_tabl_headers, pconfig={"namespace": NAMESPACE}),
+            plot = make_own_plots(),
         )
-        return data_by_sample.keys()
+        return cov_data.keys()
 
 
-def parse_wgs_coverage_metrics(f, file_regex):
-    """
+"""
+The cov_patterns list contains the regexes used to match the metrics. The regexes must be properly sorted.
+All regexes are constructed to be as general and simple as possible to speed up the matching of metrics.
+
+The "make_auto_configs" dictionary is closely related to the cov_patterns.
+The PCT_RGX and UNI_RGX are also used in the "make_user_configs".
+"""
+PCT_RGX = "^PCT of (?P<region>.+?) with coverage (?P<entity>.+)"
+AVG_RGX = "^Average (?P<entity>.+?) coverage over (?P<region>.+)"
+ALN_RGX = "^Aligned (?P<entity>.+)"
+UNI_RGX = "^Uniformity of coverage (?P<entity>.+?) over (?P<region>.+)"
+MED_RGX = "^Median (?P<entity>.+?) coverage over (?P<region>.+)"
+RAT_RGX = "(?P<entity>.+?) ratio over (?P<region>.+)"
+ANY_RGX = ".+"
```

Comment:
Please compile your regexes if you already define them in a nice way

Reply:
Do you think compiling them will improve performance? I did not test it, but i assume that speed-up would be insignificant.  These are just strings, which are used as dict keys and patterns for re.search. But if you want i can adjust the code to compile them.

Reply:
Sure, point taken, you can also decide to keep as-is. Performance would need to be measured, I don't believe this to be really warranted here. IMO this is fine.

Reply:
Actually i think you are right. These regexes are used to check incoming metrics from CSVs. It would be nice to speed-up processing of many files with lots of metrics. Especially for overloaded or weak computer. 
I'll adapt the code.